From b1d9a78f4e7d4b27df586c80f6361e14211d901a Mon Sep 17 00:00:00 2001
From: Zongdong Jiao <zongdong.jiao@amlogic.com>
Date: Mon, 29 Jun 2015 10:40:45 +0800
Subject: [PATCH 5674/5965] PD#108049: hdmitx20: reduce audio setting time

Just directly set AIU_HDMI_CLK_DATA_CTRL bit[7:4][3:0],
no nessary waste time to audio clock

Change-Id: I60a6b47e6898998d7b97f44bb9bd4aee3951b23c
---
 .../mach-mesong9bb/hdmi_tx_hw_20/hdmi_tx_hw.c | 20 -------------------
 .../mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c | 20 -------------------
 2 files changed, 40 deletions(-)

diff --git a/arch/arm/mach-mesong9bb/hdmi_tx_hw_20/hdmi_tx_hw.c b/arch/arm/mach-mesong9bb/hdmi_tx_hw_20/hdmi_tx_hw.c
index 6475211077db..6ef75ef963b1 100644
--- a/arch/arm/mach-mesong9bb/hdmi_tx_hw_20/hdmi_tx_hw.c
+++ b/arch/arm/mach-mesong9bb/hdmi_tx_hw_20/hdmi_tx_hw.c
@@ -1966,7 +1966,6 @@ static void hdmitx_setaudioinfoframe(unsigned char* AUD_DB, unsigned char* CHAN_
 void set_hdmi_audio_source(unsigned int src)
 {
     unsigned long data32;
-    unsigned int i;
 
     // Disable HDMI audio clock input and its I2S input
     data32  = 0;
@@ -1974,30 +1973,11 @@ void set_hdmi_audio_source(unsigned int src)
     data32 |= (0    << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
     aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
 
-    // Enable HDMI audio clock from the selected source
-    data32  = 0;
-    data32 |= (0    << 4);  // [5:4]    hdmi_data_sel: 00=disable hdmi i2s input; 01=Select pcm data; 10=Select AIU I2S data; 11=Not allowed.
-    data32 |= (src  << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
-    aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
-
-    // Wait until clock change is settled
-    i = 0;
-    msleep_interruptible(1000);
-    data32 = aml_read_reg32(P_AIU_HDMI_CLK_DATA_CTRL);
-    if (((data32 >> 8) & 0x3) != src)
-        printk("audio clock wait time out\n");
-
     // Enable HDMI I2S input from the selected source
     data32  = 0;
     data32 |= (src  << 4);  // [5:4]    hdmi_data_sel: 00=disable hdmi i2s input; 01=Select pcm data; 10=Select AIU I2S data; 11=Not allowed.
     data32 |= (src  << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
     aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
-
-    // Wait until data change is settled
-    msleep_interruptible(1000);
-    data32 = aml_read_reg32(P_AIU_HDMI_CLK_DATA_CTRL);
-    if (((data32 >> 12) & 0x3) != src)
-        printk("audio data wait time out\n");
 } /* set_hdmi_audio_source */
 
 static void hdmitx_set_aud_pkt_type(audio_type_t type)
diff --git a/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c b/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
index 799ab244a6fc..c069cb3eae3c 100755
--- a/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
+++ b/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
@@ -1981,7 +1981,6 @@ static void hdmitx_setaudioinfoframe(unsigned char* AUD_DB, unsigned char* CHAN_
 void set_hdmi_audio_source(unsigned int src)
 {
     unsigned long data32;
-    unsigned int i;
     
     // Disable HDMI audio clock input and its I2S input
     data32  = 0;
@@ -1989,30 +1988,11 @@ void set_hdmi_audio_source(unsigned int src)
     data32 |= (0    << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
     aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
 
-    // Enable HDMI audio clock from the selected source
-    data32  = 0;
-    data32 |= (0    << 4);  // [5:4]    hdmi_data_sel: 00=disable hdmi i2s input; 01=Select pcm data; 10=Select AIU I2S data; 11=Not allowed.
-    data32 |= (src  << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
-    aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
-    
-    // Wait until clock change is settled
-    i = 0;
-    msleep_interruptible(1000);
-    data32 = aml_read_reg32(P_AIU_HDMI_CLK_DATA_CTRL);
-    if(((data32 >> 8) & 0x3) != src)
-        printk("audio clock wait time out\n");
-
     // Enable HDMI I2S input from the selected source
     data32  = 0;
     data32 |= (src  << 4);  // [5:4]    hdmi_data_sel: 00=disable hdmi i2s input; 01=Select pcm data; 10=Select AIU I2S data; 11=Not allowed.
     data32 |= (src  << 0);  // [1:0]    hdmi_clk_sel: 00=Disable hdmi audio clock input; 01=Select pcm clock; 10=Select AIU aoclk; 11=Not allowed.
     aml_write_reg32(P_AIU_HDMI_CLK_DATA_CTRL, data32);
-
-    // Wait until data change is settled
-    msleep_interruptible(1000);
-    data32 = aml_read_reg32(P_AIU_HDMI_CLK_DATA_CTRL);
-    if(((data32 >> 12) & 0x3) != src)
-        printk("audio data wait time out\n");
 } /* set_hdmi_audio_source */
 
 static void hdmitx_set_aud_pkt_type(audio_type_t type)
-- 
2.19.0

